calicoctl/commands/node/run.go

```diff
@@ -400,10 +400,10 @@ Description:
 	if err != nil {
 		return fmt.Errorf("Error attempting to kill process: check logs for details")
 	}
+	// Wait for the logging process to terminate.  We expect an error here, because we
+	// just killed it.
 	err = logCmd.Wait()
-	if err != nil {
-		return fmt.Errorf("Error waiting for process to exit: check logs for details")
-	}
+	log.WithError(err).Info("Expected error after killing docker logs command")
```

Comment:
Does this get output to the console by default?  I think we probably _don't_ want to worry the user with this.  As it says, it's expected.

Reply:
I don't think so, but will just check...

Reply:
No, it doesn't:
```
tests.st.utils.utils: INFO: [2019-09-05 11:33:47.805158] export ETCD_ENDPOINTS=http://192.168.1.186:2379; export ETCD_CA_CERT_FILE=; export ETCD_CERT_FILE=; export ETCD_KEY_FILE=; /code/dist/calicoctl node run --node-image=calico/node:latest-amd64 --ip=192.168.1.186 --ip6=fe80::bb35:6276:9c2b:d5dc --no-default-ippools
tests.st.utils.utils: INFO:   # Running command to load modules: modprobe -a xt_set ip6_tables
tests.st.utils.utils: INFO:   # Enabling IPv4 forwarding
tests.st.utils.utils: INFO:   # Enabling IPv6 forwarding
tests.st.utils.utils: INFO:   # Increasing conntrack limit
tests.st.utils.utils: INFO:   # Removing old calico-node container (if running).
tests.st.utils.utils: INFO:   # Running the following command to start calico-node:
tests.st.utils.utils: INFO:   # 
tests.st.utils.utils: INFO:   # docker run --net=host --privileged --name=calico-node -d --restart=always -e ETCD_ENDPOINTS=http://192.168.1.186:2379 -e ETCD_DISCOVERY_SRV= -e NODENAME=glaurung -e CALICO_NETWORKING_BACKEND=bird -e NO_DEFAULT_POOLS=true -e IP=192.168.1.186 -e IP6=fe80::bb35:6276:9c2b:d5dc -v /var/log/calico:/var/log/calico -v /var/run/calico:/var/run/calico -v /var/lib/calico:/var/lib/calico -v /lib/modules:/lib/modules -v /run:/run calico/node:latest-amd64
tests.st.utils.utils: INFO:   # 
tests.st.utils.utils: INFO:   # Image may take a short time to download if it is not available locally.
tests.st.utils.utils: INFO:   # Container started, checking progress logs.
tests.st.utils.utils: INFO:   # 
tests.st.utils.utils: INFO:   # 2019-09-05 11:33:47.915 [INFO][8] logutils.go 159: Early log level set to info
tests.st.utils.utils: INFO:   # 2019-09-05 11:33:47.915 [INFO][8] logutils.go 159: Using NODENAME environment for node name
tests.st.utils.utils: INFO:   # 2019-09-05 11:33:47.915 [INFO][8] logutils.go 159: Determined node name: glaurung
tests.st.utils.utils: INFO:   # 2019-09-05 11:33:47.916 [INFO][8] logutils.go 159: Skipping datastore connection test
tests.st.utils.utils: INFO:   # 2019-09-05 11:33:47.916 [INFO][8] logutils.go 159: Building new node resource Name="glaurung"
tests.st.utils.utils: INFO:   # 2019-09-05 11:33:47.916 [INFO][8] logutils.go 159: Initialize BGP data
tests.st.utils.utils: INFO:   # 2019-09-05 11:33:47.916 [INFO][8] logutils.go 159: Using IPv4 address from environment: IP=192.168.1.186
tests.st.utils.utils: INFO:   # 2019-09-05 11:33:47.916 [INFO][8] logutils.go 159: IPv4 address 192.168.1.186 discovered on interface wlp59s0
tests.st.utils.utils: INFO:   # 2019-09-05 11:33:47.916 [INFO][8] logutils.go 159: Using IPv6 address from environment: IP6=fe80::bb35:6276:9c2b:d5dc
tests.st.utils.utils: INFO:   # 2019-09-05 11:33:47.916 [INFO][8] logutils.go 159: IPv6 address fe80::bb35:6276:9c2b:d5dc discovered on interface wlp59s0
tests.st.utils.utils: INFO:   # 2019-09-05 11:33:47.916 [INFO][8] logutils.go 159: Node IPv4 changed, will check for conflicts
tests.st.utils.utils: INFO:   # 2019-09-05 11:33:47.917 [INFO][8] logutils.go 159: No AS number configured on node resource, using global value
tests.st.utils.utils: INFO:   # 2019-09-05 11:33:47.918 [INFO][8] logutils.go 159: Skipping IP pool configuration
tests.st.utils.utils: INFO:   # 2019-09-05 11:33:47.921 [INFO][8] logutils.go 159: Using node name: glaurung
tests.st.utils.utils: INFO:   # Calico node started successfully
```
`calicoctl node run` shows calico-node's logs by default, but not its own.

Reply:
If I hack the node ST so that `calicoctl` is `calicoctl -l info`, then I see the log about the expected error:
```
tests.st.utils.utils: INFO:   # time="2019-09-05T11:44:23Z" level=info msg="Expected error after killing docker logs command" error="signal: killed"
```
Which I think is fine.